hedera-node/src/main/java/com/hedera/services/store/tokens/HederaTokenStore.java

```diff
@@ -320,7 +325,11 @@ public ResponseCodeEnum changeOwner(NftId nftId, AccountID from, AccountID to) {
 				return toFreezeAndKycValidity;
 			}
 
-			final var owner = (EntityId) nftsLedger.get(nftId, OWNER);
+			var owner = (EntityId) nftsLedger.get(nftId, OWNER);
+			if (owner.equals(fromGrpcAccountId(AccountID.getDefaultInstance()))) {
+				var tid = nftId.tokenId();
+				owner = this.tokens.get().get(new MerkleEntityId(tid.getShardNum(), tid.getRealmNum(), tid.getTokenNum())).treasury();
+			}
```

Comment:
This pattern feels like it's repeated a few places; would it be realistic to provide a `TokenStore.getEffectiveOwner(NftId)` or equivalent?

Reply:
I couldn't find any more references of this code in HederaTokenStore for owner.equals ... I think we are already doing separate refactoring PR for the stores ;)